packages/backend-data/src/factory.ts

```diff
@@ -232,6 +232,9 @@ class DataGenerator implements ConstructContainerEntryGenerator {
     });
     let amplifyApi = undefined;
 
+    const shouldProvisionHotswapFriendlyResources =
+      stableBackendIdentifiers?.getDeploymentType?.() === 'sandbox';
```

Comment:
Can you please use this instead of introducing this new API (which doesn't seem to belong to stable identifiers) ?

https://github.com/aws-amplify/amplify-backend/blob/67feca033741c8382149b26c6634fe3b28c94b8f/packages/backend-data/src/factory.ts#L268-L269

Reply:
Additionally. Can you please add unit test sfor change in `factory.ts` ?

Reply:
Thanks for your comments, Kamil. I have switched to use `isSandboxDeployment`, currently trying to figure out how to uni test this change in `factory.test.ts`. 

The image 1 below shows how I unit test this change in `amplify-category-api`, where I inspect the synthesized stack template or `CloudFormationStackArtifact` to be exact and verify through the physical resource id generated. I am struggling to do something similar in `factory.test.ts`. 

I tried `const amplifyTableStackTemplate = Template.fromStack(
      Stack.of(dataConstruct.resources.nestedStacks['Todo']));`, it looks like image 2 below. 

And `const amplifyTableStackTemplate = Template.fromStack(
      Stack.of(dataConstruct.resources.graphqlApi));` looks like image 3. 

Both are not quite what I am looking for. Also not quite sure which method of `class Template` could come in handy in this case. Could you advise on how I can unit test this change?

<img width="1076" alt="Screenshot 2024-08-22 at 9 32 43 AM" src="https://github.com/user-attachments/assets/79db8552-ba2d-465c-ad2a-ab588dec45a5">

<img width="878" alt="Screenshot 2024-08-22 at 9 53 10 AM" src="https://github.com/user-attachments/assets/5110c06c-f3b3-41a8-a72a-9fb3a2534a47">

<img width="877" alt="Screenshot 2024-08-22 at 9 54 59 AM" src="https://github.com/user-attachments/assets/514c0ee3-a464-4f7f-b724-c6e5ac63e7b6">

Reply:
Take a look at https://github.com/aws-amplify/amplify-backend/blob/97207363ec928bdc6de1ac505e98ee2e90acb941/packages/backend-data/src/factory.test.ts#L817-L828 .

Which asserts other aspect of sandbox deployment.

In `factory.test.ts` you don't have to thoroughly test data construct behaviors (this is covered by data construct tests). What you should be looking at is a single simplest signal that new flag was applied, if such signal is available you can write assertion using it.

Reply:
im not able to find it somehow. 
![image](https://github.com/user-attachments/assets/806a1272-9c62-43bd-a085-ba2443b7024b)


> a single simplest signal that new flag was applied

if i get it right, the unit test should have picked up the flag set here at line 255 right?
<img width="568" alt="Screenshot 2024-08-22 at 10 21 25 AM" src="https://github.com/user-attachments/assets/07f8d9a7-7f1d-41d1-b3dd-c2b78988777f">

Reply:
You won't find `_provisionHotswapFriendlyResources` in template. You should be looking for resources and/or their properties that are affected by that flag.

Reply:
Okay. I took a closer look at the change introduced in https://github.com/aws-amplify/amplify-category-api/pull/2772 and played with debugger a bit.

Unfortunately, it turns out that that physical resource id change can't be accessed by neither inspecting construct tree nor via template inspection. I.e. it doesn't seem to be possible.

There's yet one option to assert this. I.e. inspect physical ids in e2e tests where they can be read from CFN stacks. But it seems overkill, so let's leave this untested for now.